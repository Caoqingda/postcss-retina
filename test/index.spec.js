const postcss = require('postcss');
const fs = require('fs');
const path = require('path');

const plugin = require('../lib');


const readFile = (filePath, ...args) => new Promise((resolve, reject) => {
    fs.readFile(path.resolve(__dirname, filePath), ...args, (err, data) => {
        if (err) {
            reject();
        } else {
            resolve(data);
        }
    });
});

async function run(input, opts) {
    return postcss([plugin(opts)]).process(input)
        .then((result) => {
            expect(result.css).toMatchSnapshot();
            expect(result.warnings().length).toBe(0);
        });
}

it('no border decl or border-width unit is not "px" ', async () => {
    const input = await readFile('./case/no-border.css');
    run(input);
});

it('transform border normally', async () => {
    const input = await readFile('./case/1px-border.css');
    run(input);
});

it('transform border normally width opt 3', async () => {
    const input = await readFile('./case/1px-border.css');
    run(input);
});

it('transform border with border-radius', async () => {
    const input = await readFile('./case/border-radius.css');
    run(input);
});

it('do not transform border with pseudo', async () => {
    const input = await readFile('./case/pseudo.css');
    run(input);
});

it('transform border with repeat class', async () => {
    const input = await readFile('./case/repeat-class.css');
    run(input);
});

it('declare it is not require to transform', async () => {
    const input = await readFile('./case/no-need.css');
    run(input);
});

it('keep position whose value is not static', async () => {
    const input = await readFile('./case/position.css');
    run(input);
});
