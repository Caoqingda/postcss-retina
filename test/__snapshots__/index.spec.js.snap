// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`declare it is not require to transform 1`] = `
".foo {  
  border: 1px solid red; /*no*/ 
}

.foo1 {  
  border: 1px solid red; /*no*/ 
  border-radius: 5px;
}

.foo2 {  
  border: 1px solid red; /*no*/ 
  border-radius: 5px; /*retina*/ position: relative;
}

.foo2::before { 
  border-radius: 0.05em; 
}

.foo2::before {
  content: ' ';
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  pointer-events: none;
  box-sizing: border-box;
  font-size: 100px;
}

/* dpr is 2 */

@media only screen and (-webkit-min-device-pixel-ratio: 2), 
only screen and (min-device-pixel-ratio: 2) {
  .foo2::before {
    transform: scale(0.5);
    transform-origin: 0 0;
    width: 200%;
    height: 200%;
    font-size: 200px;
  }
}

/* dpr is 3 */

@media only screen and (-webkit-min-device-pixel-ratio: 3), 
only screen and (min-device-pixel-ratio: 3) {
  .foo2::before {
    transform: scale(0.333333);
    transform-origin: 0 0;
    width: 300%;
    height: 300%;
    font-size: 300px;
  }
}

/* dpr is 4 */

@media only screen and (-webkit-min-device-pixel-ratio: 4), 
only screen and (min-device-pixel-ratio: 4) {
  .foo2::before {
    transform: scale(0.25);
    transform-origin: 0 0;
    width: 400%;
    height: 400%;
    font-size: 400px;
  }
}"
`;

exports[`do not transform border with pseudo 1`] = `
".test:hover {
  color: red;
  position: relative;
}

.test::before {
  color: red;
}

.test::after {
  color: red;
}

.test:hover::before {
  border: 1px solid red;
}

.test:hover::before {
  content: ' ';
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  pointer-events: none;
  box-sizing: border-box;
  font-size: 100px;
}

/* dpr is 2 */

@media only screen and (-webkit-min-device-pixel-ratio: 2), 
only screen and (min-device-pixel-ratio: 2) {
  .test:hover::before {
    transform: scale(0.5);
    transform-origin: 0 0;
    width: 200%;
    height: 200%;
    font-size: 200px;
  }
}

/* dpr is 3 */

@media only screen and (-webkit-min-device-pixel-ratio: 3), 
only screen and (min-device-pixel-ratio: 3) {
  .test:hover::before {
    transform: scale(0.333333);
    transform-origin: 0 0;
    width: 300%;
    height: 300%;
    font-size: 300px;
  }
}

/* dpr is 4 */

@media only screen and (-webkit-min-device-pixel-ratio: 4), 
only screen and (min-device-pixel-ratio: 4) {
  .test:hover::before {
    transform: scale(0.25);
    transform-origin: 0 0;
    width: 400%;
    height: 400%;
    font-size: 400px;
  }
}"
`;

exports[`keep position whose value is not static 1`] = `
".test1 {
  position: static;
  position: relative;
}

.test2 {
  position: relative;
}

.test3 {
  position: absolute;
}

.test4 {
  position: fixed;
}

.test1::before {
  border: 1px solid red;
}

.test2::before {
  border: 1px solid red;
}

.test3::before {
  border: 1px solid red;
}

.test4::before {
  border: 1px solid red;
}

.test1::before, .test2::before, .test3::before, .test4::before {
  content: ' ';
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  pointer-events: none;
  box-sizing: border-box;
  font-size: 100px;
}

/* dpr is 2 */

@media only screen and (-webkit-min-device-pixel-ratio: 2), 
only screen and (min-device-pixel-ratio: 2) {
  .test1::before, .test2::before, .test3::before, .test4::before {
    transform: scale(0.5);
    transform-origin: 0 0;
    width: 200%;
    height: 200%;
    font-size: 200px;
  }
}

/* dpr is 3 */

@media only screen and (-webkit-min-device-pixel-ratio: 3), 
only screen and (min-device-pixel-ratio: 3) {
  .test1::before, .test2::before, .test3::before, .test4::before {
    transform: scale(0.333333);
    transform-origin: 0 0;
    width: 300%;
    height: 300%;
    font-size: 300px;
  }
}

/* dpr is 4 */

@media only screen and (-webkit-min-device-pixel-ratio: 4), 
only screen and (min-device-pixel-ratio: 4) {
  .test1::before, .test2::before, .test3::before, .test4::before {
    transform: scale(0.25);
    transform-origin: 0 0;
    width: 400%;
    height: 400%;
    font-size: 400px;
  }
}"
`;

exports[`no border decl or border-width unit is not "px"  1`] = `
".test1 {
  color: red;
}

.test2 {
  border: 1em red solid;
}

.test2 {
  border-style: solid;
}"
`;

exports[`transform border normally 1`] = `
".test1 {
  color: red;
  position: relative;
}

.test11 {
  color: red;
  position: relative;
}

.image {
  color: red;
  position: relative;
}

.left {
  color: red;
  position: relative;
}

.left1 {
  color: red;
  position: relative;
}

.right {
  color: red;
  position: relative;
}

.top {
  color: red;
  position: relative;
}

.bottom {
  color: red;
  position: relative;
}

.test1::before {
  border: 1px solid red;
}

.test11::before {
  border-width: 1px;
}

.image::before {
  border-image: 1px solid red;
}

.left::before {
  border-left: 1px solid red;
}

.left1::before {
  border-left-width: 1px;
}

.right::before {
  border-right: 1px solid red;
}

.top::before {
  border-top: 1px solid red;
}

.bottom::before {
  border-bottom: 1px solid red;
}

.test1::before, .test11::before, .image::before, .left::before, .left1::before, .right::before, .top::before, .bottom::before {
  content: ' ';
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  pointer-events: none;
  box-sizing: border-box;
  font-size: 100px;
}

/* dpr is 2 */

@media only screen and (-webkit-min-device-pixel-ratio: 2), 
only screen and (min-device-pixel-ratio: 2) {
  .test1::before, .test11::before, .image::before, .left::before, .left1::before, .right::before, .top::before, .bottom::before {
    transform: scale(0.5);
    transform-origin: 0 0;
    width: 200%;
    height: 200%;
    font-size: 200px;
  }
}

/* dpr is 3 */

@media only screen and (-webkit-min-device-pixel-ratio: 3), 
only screen and (min-device-pixel-ratio: 3) {
  .test1::before, .test11::before, .image::before, .left::before, .left1::before, .right::before, .top::before, .bottom::before {
    transform: scale(0.333333);
    transform-origin: 0 0;
    width: 300%;
    height: 300%;
    font-size: 300px;
  }
}

/* dpr is 4 */

@media only screen and (-webkit-min-device-pixel-ratio: 4), 
only screen and (min-device-pixel-ratio: 4) {
  .test1::before, .test11::before, .image::before, .left::before, .left1::before, .right::before, .top::before, .bottom::before {
    transform: scale(0.25);
    transform-origin: 0 0;
    width: 400%;
    height: 400%;
    font-size: 400px;
  }
}"
`;

exports[`transform border normally width opt 3 1`] = `
".test1 {
  color: red;
  position: relative;
}

.test11 {
  color: red;
  position: relative;
}

.image {
  color: red;
  position: relative;
}

.left {
  color: red;
  position: relative;
}

.left1 {
  color: red;
  position: relative;
}

.right {
  color: red;
  position: relative;
}

.top {
  color: red;
  position: relative;
}

.bottom {
  color: red;
  position: relative;
}

.test1::before {
  border: 1px solid red;
}

.test11::before {
  border-width: 1px;
}

.image::before {
  border-image: 1px solid red;
}

.left::before {
  border-left: 1px solid red;
}

.left1::before {
  border-left-width: 1px;
}

.right::before {
  border-right: 1px solid red;
}

.top::before {
  border-top: 1px solid red;
}

.bottom::before {
  border-bottom: 1px solid red;
}

.test1::before, .test11::before, .image::before, .left::before, .left1::before, .right::before, .top::before, .bottom::before {
  content: ' ';
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  pointer-events: none;
  box-sizing: border-box;
  font-size: 100px;
}

/* dpr is 2 */

@media only screen and (-webkit-min-device-pixel-ratio: 2), 
only screen and (min-device-pixel-ratio: 2) {
  .test1::before, .test11::before, .image::before, .left::before, .left1::before, .right::before, .top::before, .bottom::before {
    transform: scale(0.5);
    transform-origin: 0 0;
    width: 200%;
    height: 200%;
    font-size: 200px;
  }
}

/* dpr is 3 */

@media only screen and (-webkit-min-device-pixel-ratio: 3), 
only screen and (min-device-pixel-ratio: 3) {
  .test1::before, .test11::before, .image::before, .left::before, .left1::before, .right::before, .top::before, .bottom::before {
    transform: scale(0.333333);
    transform-origin: 0 0;
    width: 300%;
    height: 300%;
    font-size: 300px;
  }
}

/* dpr is 4 */

@media only screen and (-webkit-min-device-pixel-ratio: 4), 
only screen and (min-device-pixel-ratio: 4) {
  .test1::before, .test11::before, .image::before, .left::before, .left1::before, .right::before, .top::before, .bottom::before {
    transform: scale(0.25);
    transform-origin: 0 0;
    width: 400%;
    height: 400%;
    font-size: 400px;
  }
}"
`;

exports[`transform border with border-radius 1`] = `
".test1 {
  color: red;
  border-radius: 1px solid red;
  position: relative;
}

.test2 {
  color: red;
  border-radius: 1px;
}

.test3 {
  color: red;
  border-radius: 10px 5px 2em / 20px 25px 30%;
  position: relative;
}

.test4 {
  color: red;
  border-radius: 1px; /* retina */ position: relative;
}

.test1::before {
  border: 1px solid red;
  border-radius: 0.01em solid red;
}

.test3::before {
  border: 1px solid red;
  border-radius: 0.1em 0.05em 2em / 0.2em 0.25em 30%;
}

.test4::before {
  border-radius: 0.01em;
}

.test1::before, .test3::before, .test4::before {
  content: ' ';
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  pointer-events: none;
  box-sizing: border-box;
  font-size: 100px;
}

/* dpr is 2 */

@media only screen and (-webkit-min-device-pixel-ratio: 2), 
only screen and (min-device-pixel-ratio: 2) {
  .test1::before, .test3::before, .test4::before {
    transform: scale(0.5);
    transform-origin: 0 0;
    width: 200%;
    height: 200%;
    font-size: 200px;
  }
}

/* dpr is 3 */

@media only screen and (-webkit-min-device-pixel-ratio: 3), 
only screen and (min-device-pixel-ratio: 3) {
  .test1::before, .test3::before, .test4::before {
    transform: scale(0.333333);
    transform-origin: 0 0;
    width: 300%;
    height: 300%;
    font-size: 300px;
  }
}

/* dpr is 4 */

@media only screen and (-webkit-min-device-pixel-ratio: 4), 
only screen and (min-device-pixel-ratio: 4) {
  .test1::before, .test3::before, .test4::before {
    transform: scale(0.25);
    transform-origin: 0 0;
    width: 400%;
    height: 400%;
    font-size: 400px;
  }
}"
`;

exports[`transform border with repeat class 1`] = `
".test {
  position: relative;
}

.test {
  position: relative;
}

.test::before {
  border: 1px solid red;
}

.test::before {
  border: 1px solid red;
}

.test::before {
  content: ' ';
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  pointer-events: none;
  box-sizing: border-box;
  font-size: 100px;
}

/* dpr is 2 */

@media only screen and (-webkit-min-device-pixel-ratio: 2), 
only screen and (min-device-pixel-ratio: 2) {
  .test::before {
    transform: scale(0.5);
    transform-origin: 0 0;
    width: 200%;
    height: 200%;
    font-size: 200px;
  }
}

/* dpr is 3 */

@media only screen and (-webkit-min-device-pixel-ratio: 3), 
only screen and (min-device-pixel-ratio: 3) {
  .test::before {
    transform: scale(0.333333);
    transform-origin: 0 0;
    width: 300%;
    height: 300%;
    font-size: 300px;
  }
}

/* dpr is 4 */

@media only screen and (-webkit-min-device-pixel-ratio: 4), 
only screen and (min-device-pixel-ratio: 4) {
  .test::before {
    transform: scale(0.25);
    transform-origin: 0 0;
    width: 400%;
    height: 400%;
    font-size: 400px;
  }
}"
`;
